From 21a70e15c660002b211685740a3c505540ff6326 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Hofrichter?= <joerg.hofrichter@emerson.com>
Date: Fri, 24 Oct 2025 08:10:45 +0200
Subject: [PATCH 1/4] mpm: rfdc: applied patch to fix C++ compilation issue

This is the following patch adopted to MPM:
https://github.com/Xilinx/embeddedsw/commit/32d1f2056233cc399a4ff3147c9cc78ddb2b3c66.patch

Original author: Conall O'Griofa <conall.o'griofa@xilinx.com>
Original commit message: rfdc: Header File Clean Up

    Header clean up to:
    1. Fix C++ compilation issue
    2. Remove duplicate including of headers

    Signed-off-by: Conall O'Griofa <conall.o'griofa@xilinx.com>

    Acked-for-series: Dragan Cvetic <dragan.m.cvetic@gmail.com>
---
 mpm/include/mpm/rfdc/xrfdc.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mpm/include/mpm/rfdc/xrfdc.h b/mpm/include/mpm/rfdc/xrfdc.h
index ad2c06fa82..d1ef74cada 100644
--- a/mpm/include/mpm/rfdc/xrfdc.h
+++ b/mpm/include/mpm/rfdc/xrfdc.h
@@ -220,7 +220,8 @@
 *       cog    02/21/19 Added XRFdc_SetCalFreeze() and XRFdc_GetCalFreeze() APIs
 *       cog    04/15/19 Rename XRFdc_SetDACMode() and XRFdc_GetDACMode() APIs to
 *                       XRFdc_SetDataPathMode() and XRFdc_GetDataPathMode() respectively.
-*
+* ported from 11.1:
+*       cog    11/26/21 Header file clean up, this fixes a C++ compilation issue.
 * </pre>
 *
 ******************************************************************************/
@@ -229,6 +230,8 @@
 #ifndef RFDC_H_
 #define RFDC_H_
 
+#include <metal/device.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -245,10 +248,7 @@ extern "C" {
 #include "sleep.h"
 #endif
 #include <metal/sys.h>
-#include <metal/device.h>
 #include <metal/irq.h>
-#include <metal/atomic.h>
-#include <metal/io.h>
 #include <metal/sleep.h>
 #include "metal/alloc.h"
 #include "xrfdc_hw.h"
-- 
2.43.0

